bootstrap/viaproxy/src/main/java/org/geysermc/geyser/platform/viaproxy/GeyserViaProxyPlugin.java

```diff
@@ -0,0 +1,208 @@
+
+public class GeyserViaProxyPlugin extends ViaProxyPlugin implements GeyserBootstrap {
+
+    public static final File ROOT_FOLDER = new File(PluginManager.PLUGINS_DIR, "Geyser");
+
+    private final GeyserViaProxyLogger logger = new GeyserViaProxyLogger(LogManager.getLogger("Geyser"));
+    private GeyserViaProxyConfiguration config;
+
+    private GeyserImpl geyser;
+    private GeyserCommandManager commandManager;
+    private IGeyserPingPassthrough pingPassthrough;
+
+    @Override
+    public void onEnable() {
+        ROOT_FOLDER.mkdirs();
+
+        GeyserLocale.init(this);
+        this.onGeyserInitialize();
+
+        ViaProxy.EVENT_MANAGER.register(this);
+    }
+
+    @EventHandler
+    private void onConsoleCommand(final ConsoleCommandEvent event) {
+        final String command = event.getCommand().startsWith("/") ? event.getCommand().substring(1) : event.getCommand();
+        if (this.getGeyserCommandManager().runCommand(this.getGeyserLogger(), command + " " + String.join(" ", event.getArgs()))) {
+            event.setCancelled(true);
+        }
+    }
+
+    @EventHandler
+    private void onShouldVerifyOnlineModeEvent(final ShouldVerifyOnlineModeEvent event) {
+        final UUID uuid = event.getProxyConnection().getGameProfile().getId();
+        if (uuid == null) return;
+
+        final GeyserSession connection = GeyserImpl.getInstance().onlineConnections().stream().filter(s -> s.javaUuid().equals(uuid)).findAny().orElse(null);
+        if (connection == null) return;
+
+        if (connection.javaUsername().equals(event.getProxyConnection().getGameProfile().getName())) {
+            event.setCancelled(true);
+        }
+    }
+
+    @EventHandler
+    private void onProxyStart(final ProxyStartEvent event) {
+        this.onGeyserEnable();
+    }
+
+    @EventHandler
+    private void onProxyStop(final ProxyStopEvent event) {
+        GeyserImpl.getInstance().getSessionManager().disconnectAll("geyser.commands.reload.kick");
+        this.onGeyserDisable();
+    }
+
+    @Override
+    public void onGeyserInitialize() {
+        try {
+            final File configFile = FileUtils.fileOrCopiedFromResource(new File(ROOT_FOLDER, "config.yml"), "config.yml", s -> s.replaceAll("generateduuid", UUID.randomUUID().toString()), this);
+            this.config = FileUtils.loadConfig(configFile, GeyserViaProxyConfiguration.class);
+        } catch (IOException e) {
+            this.logger.severe(GeyserLocale.getLocaleStringLog("geyser.config.failed"), e);
+            return;
+        }
+
+        config.getRemote().setAuthType(Files.isRegularFile(this.config.getFloodgateKeyPath()) ? AuthType.FLOODGATE : AuthType.OFFLINE);
+        GeyserConfiguration.checkGeyserConfiguration(this.config, this.logger);
+
+        this.geyser = GeyserImpl.load(PlatformType.VIAPROXY, this);
+        LoopbackUtil.checkAndApplyLoopback(this.logger);
+    }
+
+    @Override
+    public void onGeyserEnable() {
+        GeyserImpl.start();
```

Comment:
Two notes:
- please reload the config when geyser is reloading, similar to how it's done here:
https://github.com/GeyserMC/Geyser/blob/master/bootstrap%2Fvelocity%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Fgeyser%2Fplatform%2Fvelocity%2FGeyserVelocityPlugin.java#L113-122

- To keep consistency reg. Geyser events, please create the command manager before starting Geyser. Additionally - I wouldn't re-create the command manager when we're reloading unless ViaProxy supports (re-)registering commands at runtime